src/DotNet/Importer.cs

```diff
@@ -358,11 +361,13 @@ public Importer(ModuleDef module, ImporterOptions options, GenericParamContext g
 
 		ITypeDefOrRef CreateTypeRef(Type type) => TryResolve(mapper?.Map(type) ?? CreateTypeRef2(type));
 
-		TypeRef CreateTypeRef2(Type type) {
+		ITypeDefOrRef CreateTypeRef2(Type type) {
+			if (TryToUseTypeDefs && IsThisModule(type.Module) && module.ResolveToken(type.MetadataToken) is TypeDef td)
```

Comment:
This operation is not entirely safe, we should check if the type is either non generic or an unbound generic type (no generic parameters provided eg. `MyType<,>`)

Reply:
This method is private, it's always called from `ImportAsTypeSig` which already check the kind of types